### PR TITLE
Add support for the CLI to push a lambda to monitor SQS queues.

### PR DESCRIPTION
**Pull request by ph**
This all support to configure triggers in the configuration file to
allow a lambda function to monitor one or more queue.

Note: we assume that body of an SQS message is a string, it we receive
a json encoded string we can use the `decode_json_fields`.

### CODE REVIEW

File: x-pack/functionbeat/provider/aws/sqs.go [hunk 1]
```diff
@@ -51,3 +82,31 @@ func (s *SQS) createHandler(client core.Client) func(request events.SQSEvent) er
...
+
+// Template returns the cloudformation template for configuring the service with the specified triggers.
+func (s *SQS) Template() *cloudformation.Template {
+	template := cloudformation.NewTemplate()
+
+	prefix := func(suffix string) string {
+		return "fnb" + s.config.Name + suffix
+	}
+
+	re := regexp.MustCompile("[:/-]")
```
Comment (urso) [inline x-pack/functionbeat/provider/aws/sqs.go]: This is kind of a constant. Let's move the init out of this function and introduce a global:

```
var reNormalizeNames = regexp.MustCompile("[:/-]")
```

It is not nice to have var as global constant, but this patterns catches errors/panics whenever the beat is started, not only when the regex is compiled by taking said code path.

A little trick (lisp-style) if ouy want to hide the regex (still creates a function pointer, though):

var normalizeSQSName = func() func(string) string {
  re := regex.MustCompile()
  return func(c string) string {
    return re.ReplaceAllString(c, "")
  }
}()

Comment (ph) [reply x-pack/functionbeat/provider/aws/sqs.go]: I have added the above method in common/string.go and tests for it.

**Comment by ph** [comment]
Updated with the reviews, added support for enabled: false in the 'yml'

**Comment by ph** [comment]
I've rebased this PR on top of the feature branch as soon as everything is green I will merge it.

**Pull request merged**